src/pages/Films/Films.tsx

```diff
@@ -1,45 +1,67 @@
 import { useEffect, useState } from "react";
 import * as S from "./Films.Styled";
 import Carousel from "../../components/Carousel/Carousel";
-import FilmsHeroImage from "../../components/FilmsBackground/FilmsBackground";
 import FilmsSearch from "../../components/FilmsSearch/FilmsSearch";
+import Toast from "../../components/Toast/Toast";
+import FilmsHeroContainer from "../../components/FilmsHeroImage/FilmsHeroImage";
 
-function Films() {
-	const [selectedGenre, setSelectedGenre] = useState(["Sci-Fi"]);
+function FilmsPage() {
+	const [selectedGenre, setSelectedGenre] = useState(["Popular"]);
 	const [fetchedFilms, setFetchedFilms] = useState([]);
-	const [heroImage, setHeroImage] = useState<string>("");
+	const [heroImage, setHeroImage] = useState<{
+		backdrop_path: string;
+		original_title: string;
+		overview: string;
+	} | null>(null);
 
-	const urlLeftSide = import.meta.env.VITE_URL_LEFT_SIDE;
-	const urlRightSide = import.meta.env.VITE_URL_RIGHT_SIDE;
+	const URL_ROOT = import.meta.env.VITE_URL_ROOT;
+	const URL_QUERY = import.meta.env.VITE_URL_QUERY;
 	const API_KEY = import.meta.env.VITE_API_KEY;
 
-	const URL = `${urlLeftSide + API_KEY + urlRightSide}`;
-
 	async function fetchData() {
+		const genreId = selectedGenre[0] === "Popular" ? "" : selectedGenre[0];
+		const URL = `${URL_ROOT + API_KEY + URL_QUERY}${
+			genreId ? `&with_genres=${genreId}` : ""
+		}`;
```

Comment:
Tutaj mamy trochę mix łączenia stringów, masz template strings ktore ulatwiaja łączenie zmiennych, ale w nich robisz plusy które stosowało się przed laty. Lepiej to ujednolicić, a nawet wyjąć do funkcji która skonstruuje nam ten url